### PR TITLE
Get intersection test

### PR DESCRIPTION
**Pull request by RomanShushakov**

### CODE REVIEW

File: app/entities/line.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from .point import Point
+from .orientation import Orientation
+
+
+class Line:
+    
+    def __init__(self, start_point: Point, end_point: Point):
+        self.start_point = start_point
```
Comment (JoacoVincent) [inline app/entities/line.py:8]: self._start_point. we use underscores for internal attributes. This change apply to all internal class attributes.

File: app/entities/line.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        self.end_point = end_point
```
Comment (JoacoVincent) [inline app/entities/line.py:9]: Please use decorator `@property` for public variables.

File: app/entities/line.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+        orientation_1 = orientation(self.start_point, self.end_point, other.start_point)
+        orientation_2 = orientation(self.start_point, self.end_point, other.end_point)
+        orientation_3 = orientation(other.start_point, other.end_point, self.start_point)
+        orientation_4 = orientation(other.start_point, other.end_point, self.end_point)
+        if (orientation_1 != orientation_2) and (orientation_3 != orientation_4): 
+            return True
+        return False
+
+    def intersection_point(self, other) -> Point:
```
Comment (JoacoVincent) [inline app/entities/line.py:28]: Empty method:
```
def intersection_point(self, other) -> Point:
    pass
```

File: app/entities/point.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class Point:
+    
+    def __init__(self, x: float, y: float):
+        self.x = x
```
Comment (JoacoVincent) [inline app/entities/point.py:4]: Add @property decorators

File: test/test_line.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
+import unittest
+from ..app.entities.point import Point
+from ..app.entities.line import Line
+
+
+class TestLine(unittest.TestCase):
+
+    @classmethod
```
Comment (JoacoVincent) [inline test/test_line.py:8]: not required

File: test/test_line.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    def setUpClass(cls):
+        print("setupClass")
+
+    @classmethod
+    def tearDownClass(cls):
+        print("teardownClass")
+
+    def setUp(self):
+        print("setUp")
```
Comment (JoacoVincent) [inline test/test_line.py:17]: do not use print()

File: test/test_line.py [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        self.assertEqual((self.line_1.end_point.x, self.line_1.end_point.y), (-5.5, 10.0))
+        self.assertEqual((self.line_2.start_point.x, self.line_2.start_point.y), (12.0, 0.0))
+        self.assertEqual((self.line_2.end_point.x, self.line_2.end_point.y), (-35.0, 17.0))
+        
+    def test_intersect(self):
+        print("test_intersect")
+        self.assertEqual(self.line_3.intersect(self.line_4), False)
+        self.assertEqual(self.line_5.intersect(self.line_6), True)
+        
+    def test_intersection_point(self):
```
Comment (JoacoVincent) [inline test/test_line.py:71]: Lets add the test only for this method

File: test/test_line.py [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+        point_17 = Point(x = 3, y = 0)
+        point_18 = Point(x = 12, y = 30)
+        self.line_9 = Line(start_point = point_17, end_point = point_18)
+        
+        point_19 = Point(x = -6, y = 3)
+        point_20 = Point(x = 11, y = 6)
+        self.line_10 = Line(start_point = point_19, end_point = point_20)
+        
+        
+    def test_coordinates(self):
```
Comment (JoacoVincent) [inline test/test_line.py:59]: remove

File: test/test_line.py [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+        print("test_coordinates")
+        self.assertEqual((self.line_1.start_point.x, self.line_1.start_point.y), (0.0, 0.0))
+        self.assertEqual((self.line_1.end_point.x, self.line_1.end_point.y), (-5.5, 10.0))
+        self.assertEqual((self.line_2.start_point.x, self.line_2.start_point.y), (12.0, 0.0))
+        self.assertEqual((self.line_2.end_point.x, self.line_2.end_point.y), (-35.0, 17.0))
+        
+    def test_intersect(self):
```
Comment (JoacoVincent) [inline test/test_line.py:66]: remove

File: app/entities/orientation.py [hunk 10]
```diff
@@ -0,0 +1,7 @@
+from enum import Enum
+
+
+class Orientation(Enum):
```
Comment (JoacoVincent) [inline app/entities/orientation.py:4]: not need int the first PR

File: app/entities/line.py [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+from .orientation import Orientation
+
+
+class Line:
+    
+    def __init__(self, start_point: Point, end_point: Point):
+        self.start_point = start_point
+        self.end_point = end_point
+
+    def intersect(self, other) -> bool:
```
Comment (JoacoVincent) [inline app/entities/line.py:11]: Not needed in the first PR

File: app/entities/line.py (hunk 3 above)
Comment (JoacoVincent) [inline app/entities/line.py:28]: Add Documentation for method

File: test/test_point.py [hunk 12]
```diff
@@ -0,0 +1,23 @@
+import unittest
+from ..app.entities.point import Point
+
+
+class TestPoint(unittest.TestCase):
+
+    @classmethod
```
Comment (JoacoVincent) [inline test/test_point.py:7]: No need setUpClass, not use print

File: app/entities/line.py (hunk 1 above)
Comment (RomanShushakov) [reply app/entities/line.py]: Done

File: app/entities/line.py (hunk 2 above)
Comment (RomanShushakov) [reply app/entities/line.py]: Done

File: app/entities/line.py (hunk 3 above)
Comment (RomanShushakov) [reply app/entities/line.py]: Done

File: app/entities/point.py (hunk 4 above)
Comment (RomanShushakov) [reply app/entities/point.py]: Done

File: test/test_line.py (hunk 5 above)
Comment (RomanShushakov) [reply test/test_line.py]: Done

File: test/test_line.py (hunk 6 above)
Comment (RomanShushakov) [reply test/test_line.py]: Done

File: test/test_line.py (hunk 7 above)
Comment (RomanShushakov) [reply test/test_line.py]: Done

File: test/test_line.py (hunk 8 above)
Comment (RomanShushakov) [reply test/test_line.py]: Done

File: app/entities/orientation.py (hunk 10 above)
Comment (RomanShushakov) [reply app/entities/orientation.py]: Done

File: app/entities/line.py (hunk 3 above)
Comment (RomanShushakov) [reply app/entities/line.py]: Done

File: test/test_point.py (hunk 12 above)
Comment (RomanShushakov) [reply test/test_point.py]: Done

File: app/entities/line.py (hunk 11 above)
Comment (RomanShushakov) [reply app/entities/line.py]: Done

File: test/testLine.py [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+        point_28 = Point(12, 6)
+        self.line_14 = Line(point_27, point_28)
+        
+    def testIntersect(self):
+        self.assertEqual(self.line_3.intersect(self.line_4), False)     # Line(Point(10, 5), Point(12, 30)) with Line((Point(-6, 3), Point(5, 5))
+        self.assertEqual(self.line_5.intersect(self.line_6), True)      # Line(Point(10, 5), Point(12, 30)) with Line((Point(-6, 3), Point(11, 6))
+        self.assertEqual(self.line_7.intersect(self.line_8), True)      # Line(Point(12, 5), Point(12, 30)) with Line((Point(-6, 3), Point(12, 6))
+        
+    def testGetIntersection(self):
+        self.assertEqual((self.line_9.getIntersection(self.line_10).x, self.line_9.getIntersection(self.line_10).y),
```
Comment (JoacoVincent) [inline test/testLine.py:35]: Use `assertAlmostEqual` for floats.

File: test/testLine.py [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+        point_27 = Point(-6, 3)
+        point_28 = Point(12, 6)
+        self.line_14 = Line(point_27, point_28)
+        
+    def testIntersect(self):
+        self.assertEqual(self.line_3.intersect(self.line_4), False)     # Line(Point(10, 5), Point(12, 30)) with Line((Point(-6, 3), Point(5, 5))
+        self.assertEqual(self.line_5.intersect(self.line_6), True)      # Line(Point(10, 5), Point(12, 30)) with Line((Point(-6, 3), Point(11, 6))
+        self.assertEqual(self.line_7.intersect(self.line_8), True)      # Line(Point(12, 5), Point(12, 30)) with Line((Point(-6, 3), Point(12, 6))
+        
+    def testGetIntersection(self):
```
Comment (JoacoVincent) [inline test/testLine.py:34]: Use `subTest` to test several test cases in one test.

File: app/entities/Line.py [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+    def start_point(self) -> Point:
+        return self._start_point
+    
+    
+    @property
+    def end_point(self) -> Point:
+        return self._end_point
+            
+            
+    def getIntersection(self, other) -> Point:
```
Comment (JoacoVincent) [inline app/entities/Line.py:22]: Use `Union`. What happened if the two lines are collinear or do not intersect.

File: app/entities/Line.py [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+        """Looking for the intersection point of two lines.
```
Comment (JoacoVincent) [inline app/entities/Line.py:23]: two line segments.

File: test/testLine.py [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+        point_25 = Point(12, 5)
+        point_26 = Point(12, 30)
+        self.line_13 = Line(point_25, point_26)
+        
+        point_27 = Point(-6, 3)
+        point_28 = Point(12, 6)
+        self.line_14 = Line(point_27, point_28)
+        
+    def testIntersect(self):
+        self.assertEqual(self.line_3.intersect(self.line_4), False)     # Line(Point(10, 5), Point(12, 30)) with Line((Point(-6, 3), Point(5, 5))
```
[the remark this answers is not in the document]
Comment (RomanShushakov) [reply test/testLine.py]: Done

File: test/testLine.py (hunk 13 above)
Comment (RomanShushakov) [reply test/testLine.py]: Done

File: test/testLine.py (hunk 14 above)
Comment (RomanShushakov) [reply test/testLine.py:26]: Done

File: app/entities/Line.py (hunk 15 above)
Comment (RomanShushakov) [reply app/entities/Line.py]: Output type has been changed to Union[Point, None]

File: app/entities/Line.py (hunk 16 above)
Comment (RomanShushakov) [reply app/entities/Line.py]: Typo has been corrected

File: test/testLine.py [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+    def setUp(self): 
+        self.input_data = [
+                (Line(Point(10.0, 5.0), Point(12.0, 30.0)), Line(Point(-6.0, 3.0), Point(11.0, 6.0))),
+                (Line(Point(3.0, 0.0), Point(12.0, 30.0)), Line(Point(-6.0, 3.0), Point(11.0, 6.0))),
+                (Line(Point(12.0, 5.0), Point(12.0, 30.0)), Line(Point(-6.0, 3.0), Point(12.0, 6.0))),
+                (Line(Point(10.0, 5.0), Point(12.0, 30.0)), Line(Point(-6.0, 3.0), Point(5.0, 5.0))),          
+                (Line(Point(12.0, 5.0), Point(16.0, 30.0)), Line(Point(12.0, 3.0), Point(16.0, 6.0))),
+            ]
+        
+        self.result = [
```
Comment (JoacoVincent) [inline test/testLine.py:17]: Test cases and results should be in the test not in the setup.

File: test/testLine.py [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+                (12.0, 6.0),
+                None, 
+                None,
+            ]
+    
+    
+    def testGetIntersection(self):
+        for i, ((line_1, line_2), res) in enumerate(zip(self.input_data, self.result)):
+            with self.subTest(i = i):
+                if i < 3:
```
Comment (JoacoVincent) [inline test/testLine.py:29]: Do not add `if` in the test. Add a new test to test `None` cases: `testGetIntersectionNone`

File: test/testLine.py (hunk 18 above)
Comment (RomanShushakov) [reply test/testLine.py]: Test cases and test results moved from setUp to testGetIntersection

File: test/testLine.py (hunk 19 above)
Comment (RomanShushakov) [reply test/testLine.py]: **testGetIntersectionNone** method has been added
